### PR TITLE
fix: Intl.PluralRulesOptions to behave according to spec

### LINKED ISSUE #33413: Intl.PluralRulesOptions does not behave according to spec
**TypeScript Version:**  3.4.0-dev.201xxxxx

**Search Terms:**
Intl.PluralRules
PluralRules

**Code**

```ts
// A *self-contained* demonstration of the problem follows...
// Test this by running `tsc` on the command-line, rather than through another build tool such as Gulp, Webpack, etc.
```

**Expected behavior:**
`Intl.PluralRules` type definition behaves according to spec
**Actual behavior:**
Missing attributes from `PluralRuleOptions`

### PR DESCRIPTION
**Pull request by longlho**
TL;DR: MDN doc is not accurate: https://github.com/tc39/ecma402/issues/365#issuecomment-530617744

According to spec, `PluralRulesOptions` do take in these extra params:
- point 2 in https://tc39.es/ecma402/#sec-intl-pluralrules-constructor) initializes the internal slot
- https://tc39.es/ecma402/#sec-initializepluralrules triggers `SetNumberFormatDigitsOptions`
- https://tc39.es/ecma402/#sec-setnfdigitoptions calls `GetNumberOption` looking for those attributes

Add `LDMLPluralRule` according to http://cldr.unicode.org/index/cldr-spec/plural-rules#TOC-Determining-Plural-Categories

ResolvedPluralRulesOptions also have these params as optional because according to spec
https://tc39.es/ecma402/#sec-initializepluralrules
https://tc39.es/ecma402/#sec-setnfdigitoptions
Current spec indicate `minimumFractionDigits` & `maximumFractionDigits` will always be there (default to 0 & 3 respectively). However in the new Unified NumberFormat spec, they can be `undefined`:
https://tc39.es/proposal-unified-intl-numberformat/section11/numberformat_diff_out.html

Fixes #33413

### CODE REVIEW

**Comment by DanielRosenwasser** [comment]
@sandersn @saschanaz would you be able to take a look?

**Comment by saschanaz** [comment]
I think we should follow the current spec rather than the new proposal, although it's already stage 3. Otherwise, LGTM.

**Comment by longlho** [comment]
The tricky part here is that
```
The following properties fall into two groups: minimumIntegerDigits, minimumFractionDigits, and maximumFractionDigits in one group, minimumSignificantDigits and maximumSignificantDigits in the other. If at least one property from the second group is defined, then the first group is ignored.
```
so `Fraction` & `Significant` are mutually exclusive.

**Comment by sandersn** [comment]
Since this missed 3.7, how about we wait until 3.8 and merge it as-is? It might advance to stage 4 in the next meeting, although I can't tell [from the agenda](https://github.com/tc39/agendas/blob/master/2019/10.md) whether this particular change will be covered with other proposal 402 business.

**Comment by longlho** [comment]
Let me change it to be current spec-compliant then. `Unified Intl.NumberFormat` is scheduled to go to stage 4 in Dec (https://docs.google.com/presentation/d/1tuW131HEuqZNd-HF6dNr-VLjNR1ItzXBaL5FloqGcbE/edit#slide=id.g442f4955c0_0_152)

I do plan to put up more diffs to fix the intl typings in general, just FYI.

**Comment by longlho** [comment]
@sandersn fixed

**Comment by sandersn** [comment]
Sorry I forgot about this for 3.8, but now's the perfect time to put things in 3.9.

**Comment by longlho** [comment]
Thanks!
